inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -753,15 +753,22 @@ void ngfunction_2_irv10(pugi::xml_node& netXml,
                 port.append_attribute("id").set_value(port_id++);
                 port.append_attribute("precision")
                     .set_value(get_precision_name(o.get_element_type()).c_str());
+
+                // Sort tensor names
+                const auto & tensor_names = o.get_tensor().get_names();
+                std::vector<std::string> vector_names(tensor_names.begin(), tensor_names.end());
+                sort(vector_names.begin(), vector_names.end());
```

Comment:
Why do we need to sort tensor names? What if we want to prioritise some names? 
Can we add new tests?

Reply:
We need to sort names because the order can be different from generation to generation. So for IR consumers it is non deterministic behavior rather than a feature.
This change was tested on full POT scope and we have tests for tensor names but l can try to add tests to check the non deterministic order but it is hard to reproduce this issue with tests since unordered_set doesn't guaranty the same order but in most cases it is the same during runs. But on MSVC the possibility to get different order is bigger, so I'll try to add some tests.